[Bug] [ProcessInstance] API queryProcessInstancesByTriggerCode not working

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.


### What happened

```text

[ERROR] 2023-07-12 20:23:13.453 +0800 org.apache.dolphinscheduler.api.exceptions.ApiExceptionHandler:[52] 
org.springframework.jdbc.BadSqlGrammarException: 
### Error querying database.  Cause: org.h2.jdbc.JdbcSQLSyntaxErrorException: Ambiguous column name "update_time"; SQL statement:
select
         
        b.id
        , b.name, b.process_definition_version, b.process_definition_code, b.project_code, b.state, b.recovery, b.start_time, b.end_time, b.run_times,host,
        command_type, b.command_param, b.task_depend_type, b.max_try_times, b.failure_strategy, b.warning_type,
        warning_group_id, b.schedule_time, b.command_start_time, b.global_params, b.flag,
        update_time, b.is_sub_process, b.executor_id, b.history_cmd,
        process_instance_priority, b.worker_group,environment_code, b.timeout, b.tenant_code, b.var_pool,
        dry_run, b.test_flag, b.next_process_instance_id, b.restart_time, b.state_history
     
        from t_ds_trigger_relation a
        join   t_ds_process_instance b on a.job_id  = b.id
        where a.trigger_type = 0 and a.trigger_code = ? [90059-210]

```

### What you expected to happen

* `queryProcessInstancesByTriggerCode` should work properly.

### How to reproduce

* Call DS `queryProcessInstancesByTriggerCode` open api.

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessInstanceMapper.xml
+++ b/dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessInstanceMapper.xml
@@ -356,6 +356,7 @@
           and next_process_instance_id = 0
     </update>
 
+<!-- There are some columns with the same names in both tables, we use a sub-query here to avoid ambiguity-->
     <select id="queryByTriggerCode" resultType="org.apache.dolphinscheduler.dao.entity.ProcessInstance">
         select
         <include refid="baseSql"/>

--- a/dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessInstanceMapper.xml
+++ b/dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessInstanceMapper.xml
@@ -28,16 +28,6 @@
         dry_run, test_flag, next_process_instance_id, restart_time, state_history
     </sql>
 
-    <sql id="baseSqlV2">
-        ${alias}.id
-        , ${alias}.name, ${alias}.process_definition_version, ${alias}.process_definition_code, ${alias}.project_code, ${alias}.state, ${alias}.recovery, ${alias}.start_time, ${alias}.end_time, ${alias}.run_times,host,
-        command_type, ${alias}.command_param, ${alias}.task_depend_type, ${alias}.max_try_times, ${alias}.failure_strategy, ${alias}.warning_type,
-        warning_group_id, ${alias}.schedule_time, ${alias}.command_start_time, ${alias}.global_params, ${alias}.flag,
-        update_time, ${alias}.is_sub_process, ${alias}.executor_id, ${alias}.history_cmd,
-        process_instance_priority, ${alias}.worker_group,environment_code, ${alias}.timeout, ${alias}.tenant_code, ${alias}.var_pool,
-        dry_run, ${alias}.test_flag, ${alias}.next_process_instance_id, ${alias}.restart_time, ${alias}.state_history
-    </sql>
-
     <select id="queryDetailById" resultType="org.apache.dolphinscheduler.dao.entity.ProcessInstance">
         select
         <include refid="baseSql"/>
@@ -368,11 +358,11 @@
 
     <select id="queryByTriggerCode" resultType="org.apache.dolphinscheduler.dao.entity.ProcessInstance">
         select
-        <include refid="baseSqlV2">
-            <property name="alias" value="b"/>
-        </include>
-        from t_ds_trigger_relation a
-        join   t_ds_process_instance b on a.job_id  = b.id
-        where a.trigger_type = 0 and a.trigger_code = #{triggerCode}
+        <include refid="baseSql"/>
+        from t_ds_process_instance a
+        join
+            (select trigger_type, job_id, trigger_code from t_ds_trigger_relation) b
+        on a.id = b.job_id
+        where b.trigger_type = 0 and b.trigger_code = #{triggerCode}
     </select>
 </mapper>

